ddtrace/contrib/tornado/stack_context.py

```diff
@@ -1,118 +1,135 @@
+import tornado
 from tornado.ioloop import IOLoop
-from tornado.stack_context import StackContextInconsistentError, _state
+import sys
 
 from ...context import Context
 from ...provider import DefaultContextProvider
 
+# tornado.stack_context deprecated in Tornado 5 removed in Tornado 6
+# instead use DefaultContextProvider with ContextVarContextManager for asyncio
+_USE_STACK_CONTEXT = not (
+    sys.version_info >= (3, 7) and tornado.version_info >= (5, 0)
+)
 
-class TracerStackContext(DefaultContextProvider):
-    """
-    A context manager that manages ``Context`` instances in a thread-local state.
-    It must be used everytime a Tornado's handler or coroutine is used within a
-    tracing Context. It is meant to work like a traditional ``StackContext``,
-    preserving the state across asynchronous calls.
-
-    Everytime a new manager is initialized, a new ``Context()`` is created for
-    this execution flow. A context created in a ``TracerStackContext`` is not
-    shared between different threads.
+if _USE_STACK_CONTEXT:
+    from tornado.stack_context import StackContextInconsistentError, _state
 
-    This implementation follows some suggestions provided here:
-    https://github.com/tornadoweb/tornado/issues/1063
-    """
-    def __init__(self):
-        # DEV: skip resetting context manager since TracerStackContext is used
-        # as a with-statement context where we do not want to be clearing the
-        # current context for a thread or task
-        super(TracerStackContext, self).__init__(reset_context_manager=False)
-        self._active = True
-        self._context = Context()
-
-    def enter(self):
-        """
-        Required to preserve the ``StackContext`` protocol.
+    class TracerStackContext(DefaultContextProvider):
```

Comment:
This is just all indented right? No changes?

Reply:
Yep, no change, just indented.